File: RedSheet.DbEntities/Models/vNanoOurObjectiveRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vNanoOurObjectiveRecords", Schema = "dbo")]
	[Serializable]
	public partial class vNanoOurObjectiveRecord  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int NanoOurObjectiveId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string NanoOurObjectiveValue { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int NanoScopeToNegotiateObjectiveId { get; set; } 

	public vNanoOurObjectiveRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-nano-our-objective-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vNanoOurObjectiveRecord {
    constructor(vNanoOurObjectiveRecord?: vNanoOurObjectiveRecord )  {
        let properties = [ "nanoOurObjectiveId", "nanoOurObjectiveValue", "nanoScopeToNegotiateObjectiveId",];
        for (let property of properties)
            if (vNanoOurObjectiveRecord && vNanoOurObjectiveRecord[property])
                this[property] = vNanoOurObjectiveRecord[property];
    }
 
	nanoOurObjectiveId : number =   0 ;
 
    @required()
    @maxLength(1000)
	nanoOurObjectiveValue : string =   undefined;
 
    @range(1,2147483647)
	nanoScopeToNegotiateObjectiveId : number =   undefined;


}
